sklearn/linear_model/huber.py

```diff
@@ -0,0 +1,349 @@
+
+    if sample_weight is None:
+        # Summing along any axis of a matrix returns a matrix object.
+        outliers_pos_sum = X_outliers_pos.sum(axis=0)
+        outliers_neg_sum = X_outliers_neg.sum(axis=0)
+
+        if X_is_sparse:
+            outliers_pos_sum = np.array(outliers_pos_sum)[0]
+            outliers_neg_sum = np.array(outliers_neg_sum)[0]
+
+        grad[:n_features] -= 2. * epsilon * outliers_pos_sum
+        grad[:n_features] += 2. * epsilon * outliers_neg_sum
+    else:
+        sw_outliers_pos = sample_weight[outliers_pos]
+        sw_outliers_neg = sample_weight[outliers_neg]
+
+        if X_is_sparse:
+            weighted_sum = (_dia_matrix(sw_outliers_pos) *
+                            X_outliers_pos).sum(axis=0)
+            weighted_sum = np.array(weighted_sum)[0]
+            grad[:n_features] -= 2. * epsilon * weighted_sum
+
+            weighted_sum = (_dia_matrix(sw_outliers_neg) *
+                            X_outliers_neg).sum(axis=0)
+            weighted_sum = np.array(weighted_sum)[0]
+            grad[:n_features] += 2. * epsilon * weighted_sum
+        else:
+            grad[:n_features] -= 2. * epsilon * (
+                np.dot(sw_outliers_pos, X_outliers_pos))
+            grad[:n_features] += 2. * epsilon * (
+                np.dot(sw_outliers_neg, X_outliers_neg))
+
+    # Gradient due to the penalty.
+    grad[:n_features] += alpha * 2. * w
+
+    # Gradient due to sigma.
+    grad[-1] = n_samples
+    grad[-1] -= n_outliers * epsilon ** 2
+    grad[-1] -= squared_loss / sigma
+
+    # Gradient due to the intercept.
+    if fit_intercept:
+        if sample_weight is None:
+            grad[-2] = -2. * np.sum(non_outliers) / sigma
+            grad[-2] -= 2. * epsilon * np.sum(outliers_pos)
+            grad[-2] += 2. * epsilon * np.sum(outliers_neg)
+        else:
+            grad[-2] = -2. * np.sum(weighted_non_outliers) / sigma
+            grad[-2] -= 2. * epsilon * np.sum(sw_outliers_pos)
+            grad[-2] += 2. * epsilon * np.sum(sw_outliers_neg)
+
+    loss = n_samples * sigma + squared_loss + outlier_loss
+    loss += alpha * np.dot(w, w)
+    return loss, grad
+
+
+class HuberRegressor(LinearModel, RegressorMixin, BaseEstimator):
+    """Linear regression model that is robust to outliers.
+
+    The Huber Regressor optimizes the squared loss for the samples where
+    |(y - X'w) / exp(sigma)| < epsilon and the absolute loss for the samples
+    where |(y - X'w) / exp(sigma)| > epsilon, where w and sigma are parameters
+    to be optimized. The parameter sigma makes sure that if y is scaled up
+    or down by a certain factor, one does not need to rescale epsilon to acheive
+    the same robustness.
+
+    This makes sure that the loss function is not heavily influenced by the
+    outliers while not completely ignoring their effect.
+
+    More specifically the loss function optimized is.
+        n_samples * exp(sigma) + (
```

Comment:
I think you need a ``math`` block here. Please check if sphinx complains and look at the rendering in html (and preferrably in pdf, too).